lab_2/main.py

```diff
@@ -90,6 +170,33 @@ def predict_language_knn(unknown_text_vector: list, known_text_vectors: list,
     :param metric: specific metric to use while calculating distance
     """
     pass
+    if not isinstance(unknown_text_vector, list) \
+            or not isinstance(known_text_vectors, list) \
+            or not isinstance(language_labels, list) \
+            or not isinstance(k, int) \
+            or not isinstance(metric, str):
+        return None
+    if len(language_labels) != len(known_text_vectors):
+        return None
+    if metric == "manhattan":
+        distances = [calculate_distance_manhattan(unknown_text_vector, vectors)
+                     for vectors in known_text_vectors]
+    elif metric == 'euclid':
+        distances = [calculate_distance(unknown_text_vector, vectors)
+                     for vectors in known_text_vectors]
+    sorted_distance = sorted(distances)[:k]
+    languages = [language_labels[distances.index(distance)] for distance in sorted_distance]
+    list_of_tuple = list(zip(languages, sorted_distance))
+    list_of_tuples = sorted(list_of_tuple, key=lambda i: i[1])
```

Comment:
`list_of_tuples` is not a very descriptive name.

Reply:
done